pkg/render/node.go

```diff
@@ -825,14 +827,23 @@ func (c *nodeComponent) cniEnvvars() []v1.EnvVar {
 func (c *nodeComponent) nodeContainer() v1.Container {
 	lp, rp := c.nodeLivenessReadinessProbes()
 	return v1.Container{
-		Name:            "calico-node",
-		Image:           c.nodeImage,
-		Resources:       c.nodeResources(),
-		SecurityContext: &v1.SecurityContext{Privileged: ptr.BoolToPtr(true)},
-		Env:             c.nodeEnvVars(),
-		VolumeMounts:    c.nodeVolumeMounts(),
-		LivenessProbe:   lp,
-		ReadinessProbe:  rp,
+		Name:      "calico-node",
+		Image:     c.nodeImage,
+		Resources: c.nodeResources(),
+		SecurityContext: &v1.SecurityContext{
+			Privileged: ptr.BoolToPtr(false),
+			RunAsUser:  ptr.Int64ToPtr(0),
+			Capabilities: &v1.Capabilities{
+				Add: []v1.Capability{
+					v1.Capability("NET_RAW"),
+					v1.Capability("NET_ADMIN"),
```

Comment:
Kernel v5.8 adds CAP_BPF.  BPF mode also needs to adjust resource limits; probably needs CAP_SYS_RESOURCE?

Reply:
Did a bit of experimenting. It appears that we still need `privileged` permissions to write to `bpffs`. We can probably adjust the permissions required if we're able to change where `tc` writes to but a quick test of setting `TC_BPF_MNT` proved unsuccessful for now. We'll probably need to tackle a reduced set of permissions for BPF at a later time.

Reply:
After some more testing, it seems that it should be possible to run BPF as non-privileged with some of the newer kernel versions (5.8+). There are a few more issues around mounts that need to be worked out, but I don't think we should block the removal of `privileged` permissions on the case anymore. We can remove privileged on BPF later.